### PR TITLE
Adds view providers for spectral plots

### LINKED ISSUE #4101: View provider for spectral plot domain object types
Spectral plots are special plots that use Wavelength or Frequency for the X-axis and any data telemetry for the Y-axis.
Currently, plots use only time for the x-axis. In order to support new views for spectral plots, we need to

Create a view provider that can display objects of type spectral plot:
1. Should only allow display of spectral plots
2. Should ensure that composition items have metadata of 'wavelength' or 'frequency' exists before allowing view of a domain object type

Spectral Aggregate plots are special plots that accept aggregate telemetry. 
Create a view provider that can display objects of type spectral plot:
1. Should only allow display of spectral aggregate plots
2. Should ensure that composition has aggregate telemetry (TBD on how to determine this)

### PR DESCRIPTION
**Pull request by shefalijoshi**
Resolves #4101 

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?

### Author Checklist

* [X] Changes address original issue?
* [X] Unit tests included and/or updated with changes?
* [X] Command line build passes?
* [X] Has this been smoke tested?
* [X] Testing instructions included in associated issue?

### CODE REVIEW

File: example/generator/plugin.js [hunk 1]
```diff
@@ -61,6 +63,25 @@ define([
...
+        openmct.types.addType("example.spectral-generator", {
+            name: "Spectral Generator",
+            description: "For development use. Generates example streaming telemetry data using a simple sine wave algorithm.",
+            cssClass: "icon-generator-telemetry",
+            creatable: true,
+            initialize: function (object) {
+                object.telemetry = {
+                    period: 10,
+                    amplitude: 1,
+                    wavelength: 1,
```
Comment (scottbell) [inline example/generator/plugin.js:75]: do we need frequency here too, or is it just derived?

File: src/plugins/plot/spectralPlot/SpectralPlotCompositionPolicy.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        }
+
+        let metadata = openmct.telemetry.getMetadata(domainObject);
+
+        return metadata.values().length > 0 && hasSpectralDomainAndRange(metadata);
+    }
+
+    return {
+        allow: function (parent, child) {
+            return ((parent.type === 'telemetry.plot.spectral') && (hasSpectralTelemetry(child)));
```
Comment (scottbell) [inline src/plugins/plot/spectralPlot/SpectralPlotCompositionPolicy.js:26]: i had trouble with this logic being overly broad. namely, it prevented any composition!

**Review by scottbell — CHANGES_REQUESTED** [review]

File: example/generator/GeneratorProvider.js [hunk 3]
```diff
@@ -86,6 +86,7 @@ define([
         var workerRequest = this.makeWorkerRequest(domainObject, request);
         workerRequest.start = request.start;
         workerRequest.end = request.end;
+        workerRequest.spectra = request.spectra;
```
Comment (shefalijoshi) [inline example/generator/GeneratorProvider.js:89]: Remind me why we need this?

File: example/generator/plugin.js (hunk 1 above)
Comment (shefalijoshi) [reply example/generator/plugin.js:75]: Good point. Let's leave it at Wavelength for now. We won't be able to use Spectral Generator based on frequency, but I think that's ok.

**Review by shefalijoshi — COMMENTED** [review]
Excellent! Thank you!

**Comment by shefalijoshi** [comment]
@scottbell Please resolve the conflict and we can merge this.

**Pull request merged**